api/src/test/java/jakarta/data/displaynamegeneration/test/ReplaceCamelCaseAndUnderscoreTest.java

```diff
@@ -0,0 +1,30 @@
+package jakarta.data.displaynamegeneration.test;
+
+import jakarta.data.displaynamegeneration.ReplaceCamelCaseAndUnderscore;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+@DisplayNameGeneration(ReplaceCamelCaseAndUnderscore.class)
+class ReplaceCamelCaseAndUnderscoreTest {
+    private ReplaceCamelCaseAndUnderscore replaceCamelCaseAndUnderscore;
+
+    @BeforeEach
+    void setUp() {
+        replaceCamelCaseAndUnderscore = new ReplaceCamelCaseAndUnderscore();
+    }
+
+    @Test
+    void shouldManageCamelCaseAndUnderscoreVeryWell() {
```

Comment:
Here are some missing test cases: 

1. Numbers should be considered words: `shouldReturn5Errors` -> `Should return 5 errors`
2. What if there is no ending underscore such as at the end of a sentence: `shouldReturnTheValueOf_maxResults` -> `Should return the value of maxResults`

Reply:
Currently, the 2nd test is green because the underscore is considered as an opening underscore even without a closing one.

```java
        if (input.contains("_")) {
            boolean insideUnderscores = false;
            for (int i = 1; i < input.length(); i++) {
                char currentChar = input.charAt(i);
                if (currentChar == '_') {
                    result.append(' ');
                    /*
                     * If the current char is an underscore and insideUnderscores is true,
                     * it means there is an opening underscore and this one is the closing one
                     * then we set insideUnderscores to false.
                     * */
                    /*
                     * If the current char is an underscore and insideUnderscores is false,
                     * it means there is not an opening underscore and this one is the opening one
                     * then we set insideUnderscores to true.
                     * */
                    insideUnderscores = !insideUnderscores;

```

Reply:
Now, I am trying to figure out the 1st test

Reply:
Done